### PR TITLE
 chore(deps): Upgrade hapi/joi dependency

### PR DESCRIPTION
**Pull request by xlisachan**
## Because

- the hapi/joi dependency is deprecated and several versions of it was used throughout the codebase

## This pull request

- [x] upgrades hapi/joi dependency to joi v17.4.0 (note: while joi v17.6.0 is the latest version, upgrading to joi 17.6.0 causes a "Cannot mix different versions of joi schemas" since `celebrate` uses joi 17.4.0.)
- [x] updates outdated joi usage - confirmed with Reino that there is no need to revise where `error !== null` as "when `fxa-payments-server` gets an error from the backend, it throws an APIError"
- [x] [updates hapi-swagger and its plugins](https://github.com/glennjones/hapi-swagger#installation)

## Issue that this pull request solves

Closes: #12178 
Closes: #12477 
Closes: #12528 

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.

## Follow-up tickets
-[ Investigate `typesafe-joi` dependency and better solution as it is a fork of hapi/joi v15.x](https://mozilla-hub.atlassian.net/browse/FXA-5066)

### CODE REVIEW

File: packages/fxa-auth-server/test/oauth/routes/token.js [hunk 1]
```diff
@@ -49,68 +48,54 @@ describe('/token POST', function () {
...
-        {
-          client_secret: CLIENT_SECRET,
-          code: CODE,
-        },
-        (err) => {
-          joiRequired(err, 'client_id');
-          done();
-        }
-      );
+    it('fails with no client_id', async () => {
```
Comment (dschom) [inline packages/fxa-auth-server/test/oauth/routes/token.js:55]: 'async' not needed

**Review by dschom — APPROVED** [review]
LGTM! Just remove that one unneded async statement and it is good to go.

File: packages/fxa-auth-server/test/oauth/routes/token.js (hunk 1 above)
Comment (xlisachan) [reply packages/fxa-auth-server/test/oauth/routes/token.js]: Removed, thanks Dan!!

**Comment by xlisachan** [comment]
Please note, the first commit (`Upgrades joi, hapi-swagger and its plugins`) has been approved by Dan.

The second commit (`Rewrite typesafe-joi and remove it from package.jsons; still exists in fxa-event-broker`) needs to be reviewed.

Thanks! :)

File: packages/fxa-event-broker/package.json [hunk 2]
```diff
@@ -60,6 +60,7 @@
     "google-auth-library": "^8.0.2",
     "graphql": "^15.6.1",
     "hot-shots": "^9.0.0",
+    "joi": "17.4.0",
```
Comment (dschom) [inline packages/fxa-event-broker/package.json:63]: I just noticed that @hapi/joi is still referenced is still a dependcy above. Is there a reason for this?

Comment (xlisachan) [reply packages/fxa-event-broker/package.json:63]: `fxa-event-broker` still has `typesafe-joi` as a dependency, which requires a peer dependency of `hapi/joi@15.x`. When I remove hapi/joi, I find this during `yarn install`:

`➤ YN0002: │ fxa-event-broker@workspace:packages/fxa-event-broker doesn't provide @hapi/joi (p86a06), requested by typesafe-joi`

We have a [ticket](https://mozilla-hub.atlassian.net/browse/FXA-5066) to investigate `typesafe-joi`, but I think we can rewrite it entirely as it has been done in `fxa-shared` in this PR. I'd like to throw it into the next sprint.

Comment (dschom) [reply packages/fxa-event-broker/package.json:63]: Okay, sounds good. I saw the changes for typesafe-joi in fxa-shared and didn't now there was more to it. Thanks for filing a follow up!

**Comment by xlisachan** [comment]
@dschom, please let me know if you have any additional feedback or whether it's good to merge as I don't believe you can mark this PR as approved again after your initial review. Thanks! :)

**Comment by xlisachan** [comment]
(Dan said it's gtg, rebasing and merging! \o/)

**Pull request merged**